app/models/physical_storage.rb

```diff
@@ -46,6 +47,41 @@ def refresh_ems
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
+  def raw_delete_physical_storage
+    raise NotImplementedError, _("raw_delete_physical_storage must be implemented in a subclass")
+  end
+
+  def delete_physical_storage
+    raw_delete_physical_storage
+  end
+
+  supports :delete do
```

Comment:
So you already have `supports_not :delete` above, you don't need this block to say that delete isn't supported